website/admin/views/__init__.py

```diff
@@ -12,20 +12,24 @@ def user(user_id):
     return render_template("internal/home.html", admin=True, rooms=rooms, user_name=user_info["name"])
 
 
-@admin_site.route('/faulty_devices', methods=['GET'])
-def faulty_devices():
-    faulty_devices = data_interface.get_faulty_devices()
-    devices = []
-    for d in faulty_devices:
-        house = data_interface.get_house_info(d['house_id'])
-        user = data_interface.get_user_info(house['user_id'])
-        devices.append({"device": d, "user": user})
-    return render_template("admin/faulty_devices.html", devices=devices)
+@admin_site.route('/fault_status', methods=['GET'])
+def fault_status():
+    fault_status = data_interface.get_admin_fault_status()
+    int_list = []
+    for one_device in fault_status:
+        int_list.append(one_device['faulty'])
+    faults_list = [(k, len(list(v))) for k, v in itertools.groupby(sorted(int_list))]
```

Comment:
What do you want to do here? This syntax is not really clear and I'm not sure it does what you intend it to do?